File: reto03-two-sum-II-input-array-is-sorted/javascript/twoSum2.js
function main() {
  console.log(twoSum2([2, 7, 11, 15], 9)); // [1, 2]
  console.log(twoSum2([2, 3, 4], 6)); // [1, 3]
  console.log(twoSum2([-1,0], -1)); // [1, 2]
}

function twoSum2(numbers, target) {
  let start = 0;
  let end = numbers.length - 1;
  let resp = null;
  while (true) {
    let sum = numbers[start] + numbers[end];
    if (sum == target) {
      resp = [start + 1, end + 1];
      break;
    }

    if (sum < target) {
      start++;
    }

    if (sum > target) {
      end--;
    }

    if (start == end) {
      break;
    }
  }
  return resp; 
}

main();

File: reto03-two-sum-II-input-array-is-sorted/java/TwoSum2.java
import java.util.Map;
import java.util.HashMap;

class TwoSum2 {
  public static void main(String[] args) {
    System.out.println(printArr(twoSum2(new int[] { 2, 7, 11, 15 }, 9)));
    System.out.println(printArr(twoSum2(new int[] { 2, 3, 4 }, 6)));
    System.out.println(printArr(twoSum2(new int[] { -1, 0 }, -1)));
  }

  public static int[] twoSum2(int[] numbers, int target) {
    int start = 0;
    int end = numbers.length - 1;
    int[] resp = null;
    while (true) {
      int sum = numbers[start] + numbers[end];
      if (sum < target) {
        start++;
      }

      if (sum > target) {
        end--;
      }

      if (sum == target) {
        resp = new int[] { start + 1, end + 1 };
        break;
      }

      if (start == end) {
        break; 
      }
    }
    return resp;
  }

  public static String printArr(int[] arr) {
    String resp = "";
    resp += "[";
    for (int i = 0; i < arr.length; i++) {
      resp += arr[i];
      if (i != arr.length - 1) {
        resp += ", ";
      }
    }
    resp += "]";
    return resp;
  }
}
